pkg/jobstore/boltdb/store.go

```diff
@@ -1050,11 +1134,157 @@ func (b *BoltJobStore) deleteJob(
 			return err
 		}
 	}
+
+	// Remove from job names bucket
+	jobNameKey := b.generateJobNameKey(job.Name, job.Namespace)
+	jobNamesBucket, err := NewBucketPath(BucketJobsNamesIndex).Get(tx, false)
```

Comment:
From a backward compatibility , fail gracefully when the job name does not exist

Reply:
Versions bucket is name-spaced under the job bucket, will be deleted with it.